includes/class-amp-theme-support.php

```diff
@@ -331,6 +333,37 @@ public static function add_meta_viewport() {
 		echo '<meta name="viewport" content="width=device-width,minimum-scale=1">';
 	}
 
+	/**
+	 * Print analytics data.
+	 */
+	public static function add_analytics_data() {
+		$analytics_entries = apply_filters( 'amp_post_template_analytics', array() );
```

Comment:
The other instances of the `amp_post_template_analytics` filter being applied are passing a `$post` as its second argument. This would be needed when a plugin needs to vary the analytics by the current post. Maybe this should, to maximize back-compat with existing plugins that filter `amp_post_template_analytics` do something like this:

```php
$analytics_entries = amp_add_custom_analytics( array() );
if ( is_singular() ) {
    $analytics_entries = apply_filters( 'amp_post_template_analytics', $analytics_entries, get_queried_object() );
}
```

For that matter, the current `amp_add_custom_analytics()` is somewhat bizzare in `amp.php`. Maybe it would be better to add another helper function to `amp-helper-functions.php` like `amp_get_analytics()` which then contains the body of what `amp_add_custom_analytics()` has today. Then at the end of that function, it can apply the `amp_post_template_analytics` filter per above. The `amp_get_analytics()` function could then be used in `\AMP_Post_Template::__construct()` instead of manually applying the `amp_post_template_analytics` filter there.

This could then be removed from `amp.php`:

```php
add_filter( 'amp_post_template_analytics', 'amp_add_custom_analytics' );
```